File: GitDesktop/GitDesktop/Models/UncommittedChangesStrategy.swift
import Foundation

/// Strategy for handling uncommitted changes on branch switch.
/// Port of `models/uncommitted-changes-strategy.ts`.
public enum UncommittedChangesStrategy: String, Codable, Sendable {
    case askForConfirmation = "AskForConfirmation"
    case stashOnCurrentBranch = "StashOnCurrentBranch"
    case moveToNewBranch = "MoveToNewBranch"
}

public let defaultUncommittedChangesStrategy: UncommittedChangesStrategy = .askForConfirmation

File: electron/app/src/models/uncommitted-changes-strategy.ts
export enum UncommittedChangesStrategy {
  AskForConfirmation = 'AskForConfirmation',
  StashOnCurrentBranch = 'StashOnCurrentBranch',
  MoveToNewBranch = 'MoveToNewBranch',
}

export const defaultUncommittedChangesStrategy: UncommittedChangesStrategy =
  UncommittedChangesStrategy.AskForConfirmation
